src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -624,30 +625,42 @@ def _create_venv_script(
             #!{bash.path}
             set -euo pipefail
 
-            # N.B.: We convert all sandbox root relative paths to absolute paths so this script
-            # works when run with a cwd set elsewhere.
-
             # N.B.: This relies on BASH_SOURCE which has been available since bash-3.0, released in
-            # 2004. In turn, our use of BASH_SOURCE relies on the fact that this script is executed
-            # by the engine via its absolute path.
-            ABS_SANDBOX_ROOT="${{BASH_SOURCE%/*}}"
+            # 2004. It will either contain either the absolute path of the venv script or the
+            # relative path from the CWD to the venv script. Either way, we know the venv script
+            # path parent directory is the sandbox root directory.
+            SANDBOX_ROOT="${{BASH_SOURCE%/*}}"
 
-            function ensure_absolute() {{
+            function adjust_relative_paths() {{
```

Comment:
OK - hopefully this is now becoming crystal clear for everyone. The only missing detail is a very core thing in a POSIX system: there is no requirement for absolute or relative file paths in general, these are only imposed by third party software if at all. All filesystem operations always account for CWD and if a path is absolute, ignore it or if a path is relative, their resolved by joining with CWD. So `absolute` was a general conceptual red-herring of a requirement here.